### ISSUE TITLE
[HOLD for payment 2024-06-05] [HOLD for payment 2024-05-06] [$250] Android - Room - User able to add VOIP number in WS but if added in room, contact disappears

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#39495]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.59
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://expensify.testrail.io/index.php?/tests/view/4472225
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. Launch app
2. Tap profile-- Workspaces---Workspace
3. Tap members
4. Tap invite
5. Enter an VOIP number and invite the user
6. Note contact is added
7. Navigate to KHN
8. Tap on a room
9. Tap header--- members --- invite
10. Enter an VOIP number and invite the user

## Expected Result:
User able to add VOIP number in WS. Similarly, the number should be able to be added in room also as a member

## Actual Result:
User able to add VOIP number in Workspace but same number if added in room, contact disappears

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/1b8d5426-212e-4ab2-9bca-ff9268a796aa

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~019af136035d7a8e29</li>
        <li>Upwork Job ID: 1777380774047997952</li>
        <li>Last Price Increase: 2024-04-08</li>
<li>Automatic offers: </li>
<ul>
<li>akinwale | Reviewer | 0</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @adelekennedy / @adelekennedy</details>

**Comment by lanitochka17** [comment]
We think that this bug might be related to #vip-vsp

**Comment by adelekennedy** [comment]
Let me check in the product room if we want to allow VOIP numbers (BE change) or, as I suspect we _don't_ want to allow them and so we should add an error message when someone tries to invite a voip number

**Comment by adelekennedy** [comment]
asked more about this in Product

**Comment by adelekennedy** [comment]
We don't support voip numbers and won't going forward - @GandalfGwaihir from your comment I think we should make this external and throw an error when the admin initially tries to add the new user

**Comment by adelekennedy** [comment]
@GandalfGwaihir can you expand on that? Would an admin be able to add a VOIP number (no errors shown) but that number would not show up in the workspace search results or room search results?

**Comment by adelekennedy** [comment]
@akinwale looks like there is one proposal to review when you can!

**Comment by akinwale** [comment]
We can move forward with @nkdengineer's [proposal ](https://github.com/Expensify/App/issues/39495#issuecomment-2045422039)here.

🎀👀🎀 C+ reviewed.

**Comment by adelekennedy** [comment]
chill melvin - we are waiting on the the PR

**Comment by nkdengineer** [comment]
I'm working on the PR, will open this today.

**Comment by nkdengineer** [comment]
@akinwale The PR is [here](https://github.com/Expensify/App/pull/40272).

**Comment by akinwale** [comment]
@adelekennedy Based on https://github.com/Expensify/App/pull/40272#issuecomment-2060474191, do we move forward with the current implementation outlined as follows?

1. When a VoIP number is added to the workspace, the action is successful.
2. When the same VoIP number is invited to a chat within the workspace, an error is displayed.

**Comment by adelekennedy** [comment]
@puneetlath for your thoughts here as well - i would think _both_ should be unsuccessful and display an error because we made the choice not to support VoIP numbers, but it's more critical that when  that errors is invited to a chat the error is displayed because that's here the use is more likely to take action

**Comment by puneetlath** [comment]
Interesting. I would think first off what we should do is make sure that the client properly handles whatever the back-end returns. So if the back-end request errors out, we need to handle that. If it succeeds, we need to handle that. 

Then, as a follow-up, we can look into whether we want to change the back-end behavior as well. Sound good?

**Comment by akinwale** [comment]
@puneetlath The [PR](https://github.com/Expensify/App/pull/40272) correctly handles the mentioned behaviours. So should we move forward as is and then determine changing the backend behaviour in a separate issue?

**Comment by adelekennedy** [comment]
@puneetlath sure! What is the question specifically? From [this thread](https://expensify.slack.com/archives/C03U7DCU4/p1712593165488579) it sounds like we block VOIP numbers for account creation and we agreed we should block in this case as well

**Comment by puneetlath** [comment]
Yeah, I guess I'm confused by that thread. It says that we block them during account creation, but we don't actually seem to. So let's create an Internal issue to make sure that happens.

**Comment by nkdengineer** [comment]
@akinwale The PR v2 is [here](https://github.com/Expensify/App/pull/41479). The bug from https://github.com/Expensify/App/issues/41133 happens after my PR because the inconsistent way we do to get the participant of the room in the detail page and member page

1. In `ReportDetailPage`, we get the participant via `ReportUtils.getVisibleChatMemberAccountIDs` function which will get the member via `participants` field of the report

https://github.com/Expensify/App/blob/729c9ac733e721132b7b9792e6c3e146e8eac41c/src/pages/ReportDetailsPage.tsx#L92

2. Meanwhile in `RoomMemberPage`, we're getting member via `report?.visibleChatMemberAccountIDs` and when we open this page again, `openRoomMembersPage` is called which will update `report.visibleChatMemberAccountIDs` without error member and then this member disappear.


So I think we can fix https://github.com/Expensify/App/issues/41133 as a follow-up issue, not a regression. cc @puneetlath

**Comment by puneetlath** [comment]
Ok, sounds good.

**Comment by puneetlath** [comment]
Bump @akinwale @nkdengineer @adelekennedy is there anything left to do here?

**Comment by adelekennedy** [comment]
@nkdengineer it looks like https://github.com/Expensify/App/issues/41133 was merged and deployed - what do we need to do with this issue? Payment or are their additional steps?

**Comment by nkdengineer** [comment]
The previous PR is reverted. Waiting https://github.com/Expensify/App/pull/41479 is merged and then we can process the next step.

**Comment by adelekennedy** [comment]
merge freeze is still in effect - is it possible to work on the next step before it's merged or do we need to wait until the freeze is lifted?

**Comment by puneetlath** [comment]
Asked about merging here: https://expensify.slack.com/archives/C07J32337/p1716491661640639

**Comment by adelekennedy** [comment]
Payouts due:

- [ ] Contributor: $250 @nkdengineer (Upwork) @nkdengineer I extended a [manual offer](https://www.upwork.com/nx/wm/offer/102634708) I'll pay out once you accept
- [ ] Contributor+: $250 @akinwale (Upwork)

Upwork job is [here](https://www.upwork.com/ab/applicants/1777380774047997952/hired).

**Comment by nkdengineer** [comment]
Hi @adelekennedy this issue was created before Apr 5 when the new price change takes effect, so according to [this annoucement](https://expensify.slack.com/archives/C01GTK53T8Q/p1712256409152489) it should have a base price of $500. Could you help to update the offer accordingly?

Thank you 🙇

**Comment by adelekennedy** [comment]
🤦‍♀️ thank you!

**Comment by nkdengineer** [comment]
@adelekennedy Thanks! I accepted the offer
